walletclient.go

```diff
@@ -1,117 +1,106 @@
+// This method is useful for scenarios where the client needs to authenticate using a less sensitive key than an xPriv.
+// - `accessKey`: The access key used for API authentication.
+// - `serverURL`: The URL of the server the client will interact with.
+func NewWithAccessKey(accessKey, serverURL string) *WalletClient {
+	return newWalletClient(
+		&AccessKey{AccessKeyString: &accessKey},
+		&HTTP{ServerURL: &serverURL},
+		&SignRequest{Sign: Ptr(true)},
+	)
+}
 
-	transportOptions := make([]transports.ClientOps, 0)
-	if client.xPriv != nil {
-		transportOptions = append(transportOptions, transports.WithXPriv(client.xPriv))
-		transportOptions = append(transportOptions, transports.WithXPub(client.xPub))
-	} else if client.xPub != nil {
-		transportOptions = append(transportOptions, transports.WithXPub(client.xPub))
-	} else if client.accessKey != nil {
-		transportOptions = append(transportOptions, transports.WithAccessKey(client.accessKey))
-	}
-	if len(client.transportOptions) > 0 {
-		transportOptions = append(transportOptions, client.transportOptions...)
-	}
+// newWalletClient creates a new WalletClient using the provided configuration options.
+func newWalletClient(configurators ...WalletClientConfigurator) *WalletClient {
+	client := &WalletClient{}
 
-	if client.transport, err = transports.NewTransport(transportOptions...); err != nil {
-		return nil, err
+	for _, configurator := range configurators {
+		configurator.Configure(client)
 	}
 
-	client.TransportService = client.transport
-
-	return client, nil
+	return client
 }
 
-// SetAdminKey set the admin key to use to create new xpubs
-func (b *WalletClient) SetAdminKey(adminKeyString string) error {
-	adminKey, err := bip32.NewKeyFromString(adminKeyString)
-	if err != nil {
-		return err
+// processMetadata will process the metadata
+func processMetadata(metadata *models.Metadata) *models.Metadata {
+	if metadata == nil {
+		m := make(models.Metadata)
+		metadata = &m
 	}
 
-	b.transport.SetAdminKey(adminKey)
-
-	return nil
+	return metadata
 }
 
-// SetSignRequest turn the signing of the http request on or off
-func (b *WalletClient) SetSignRequest(signRequest bool) {
-	b.transport.SetSignRequest(signRequest)
+// addSignature will add the signature to the request
+func addSignature(header *http.Header, xPriv *bip32.ExtendedKey, bodyString string) ResponseError {
+	return setSignature(header, xPriv, bodyString)
 }
 
-// IsSignRequest return whether to sign all requests
-func (b *WalletClient) IsSignRequest() bool {
-	return b.transport.IsSignRequest()
+func Ptr[T any](obj T) *T {
+	return &obj
 }
 
-// GetTransport returns the current transport service
-func (b *WalletClient) GetTransport() *transports.TransportService {
-	return &b.transport
+// getPrivKey retrieves the client's private key. If the primary key is not set,
+func (wc *WalletClient) getPrivKey() *bip32.ExtendedKey {
```

Comment:
Sorry, but I don't like the idea of this method:
1. it's misleading when reading a usage, because we have separated fields for admin key and xpriv, I thought that we almost always use wrong key
2. we're not providing the feature (which is in js-client) to create single client with both admin and user access. Maybe for some users, or for sure for scripts it would be more convenient to have it configured like that.